irony-cdb-libclang.el

```diff
@@ -0,0 +1,65 @@
+;;; irony-cdb-libclang.el --- Compilation Database for irony using libclang
+
+;; Copyright (C) 2014  Guillaume Papin
+;;                     Karl Hylén
+
+;; Author: Guillaume Papin <guillaume.papin@epitech.eu>
+;;         Karl Hylén <karl.hylen@gmail.com>
+;; Keywords: c, convenience, tools
+
+;; This program is free software; you can redistribute it and/or modify
+;; it under the terms of the GNU General Public License as published by
+;; the Free Software Foundation, either version 3 of the License, or
+;; (at your option) any later version.
+
+;; This program is distributed in the hope that it will be useful,
+;; but WITHOUT ANY WARRANTY; without even the implied warranty of
+;; MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+;; GNU General Public License for more details.
+
+;; You should have received a copy of the GNU General Public License
+;; along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+;;; Commentary:
+;;
+;; Compilation Database support for Irony using libclangs CXCompilationDatabase,
+;; http://clang.llvm.org/doxygen/group__COMPILATIONDB.html
+;;; Code:
+
+(require 'irony-cdb)
+(require 'irony-cdb-json)
+
+(require 'cl-lib)
+
+;;;###autoload
+(defun irony-cdb-libclang (command &rest args)
+  (cl-case command
+    (get-compile-options (irony-cdb-libclang--get-compile-options))))
+
+(defun irony-cdb-libclang--get-compile-options ()
+  (irony--awhen (irony-cdb-json--locate-db)
+    (irony-cdb-libclang--server-exact-flags it)))
+
+(defun irony-cdb-libclang--server-exact-flags (db-file)
+  "Get compile options from server"
+  (let ((project-root (file-name-directory db-file))
```

Comment:
I agree and disagree.

`db-dir` is right in practice.

In theory, the Clang functions takes the build directory:

* http://clang.llvm.org/docs/JSONCompilationDatabase.html#build-system-integration

  > The convention is to name the file compile_commands.json and put it at the top of the build directory. Clang tools are pointed to the top of the build directory to detect the file and use the compilation database to parse C++ code in the source tree.

* http://clang.llvm.org/doxygen/group__COMPILATIONDB.html#gae0822a6a54afaea92bad5d3b3256bf26

I'm fiine with db-dir.